pkg/osquery/tables/macos_software_update/recommended_updates_table.go

```diff
@@ -0,0 +1,81 @@
+//go:build darwin
+// +build darwin
+
+package macos_software_update
+
+/*
+#cgo darwin CFLAGS: -DDARWIN -x objective-c
+#cgo darwin LDFLAGS: -framework Cocoa
+#include "sus.h"
+*/
+import "C"
+import (
+	"context"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/launcher/pkg/dataflatten"
+	"github.com/kolide/launcher/pkg/osquery/tables/dataflattentable"
+	"github.com/kolide/launcher/pkg/osquery/tables/tablehelpers"
+	"github.com/osquery/osquery-go/plugin/table"
+)
+
+var updatesData []map[string]interface{}
```

Comment:
Does this need to be a package level variable? I can't tell if it needs that for how the c functions work, or not.

If it does, how confident are you that it's reset between generate calls?

Reply:
I could have avoided it being package level and moved it to `getUpdates` but then I would need to pass a pointer to it into the C code, then pass it back to Go as an argument in the callbacks, cast it back to it's Go type, etc. Although I prefer to scope var as narrowly as possible, this felt more simple.

When `generate` is first called, this var will be in whatever state is was left in since the last invocation. When `updatesFound` is called, it will reinitialize the var. However, there is an edge case where if the second invocation returns no results in the C code, `updatesFound` won't get called, the var doesn't get reset, and the stale value remains and gets returned. I can add an explicit reset to make sure the stale value is always tossed before subsequent invocations.

Reply:
I think an explicit reset would be good. Either at the start of the `generate` or in a defer. Whatever feels right.